src/test/fuzz/util/descriptor.cpp

```diff
@@ -84,3 +87,34 @@ bool HasDeepDerivPath(const FuzzBufferType& buff, const int max_depth)
     }
     return false;
 }
+
+bool HasLargeFrag(const FuzzBufferType& buff, const int max_subs)
+{
+    std::stack<int> counts;
+    for (const auto& ch: buff) {
+        if (ch == '(') {
+            counts.push(1);
+        } else if (ch == ',' && !counts.empty()) {
+            if (++counts.top() > max_subs) return true;
+        } else if (ch == ')' && !counts.empty()) {
+            counts.pop();
+        }
+    }
+    return false;
+}
+
+bool HasTooManyWrappers(const FuzzBufferType& buff, const int max_wrappers)
+{
+    auto count{0};
+    // We iterate in reverse order to start counting when we encounter a colon.
+    for (const auto ch: reverse_iterate(buff)) {
+        if (ch == ':') {
+            count++;
+        } else if (ch == ',' || ch == '(' || ch == '{') {
```

Comment:
I can't find any reference to fragments starting with '{' in the [BIP](https://github.com/bitcoin/bips/blob/master/bip-0379.md), is `|| ch == '{'` necessary?

Reply:
Taproot leaves. It's to avoid overcounting if for instance you have `tr(d971308d5463ddc82addd897f6ebfa8a88e86de0664a66b9e9cf5e873459c528,{dvdvdvdvdv:1,1))`.

Reply:
Okay, thanks. No strong opinion as to which approach, but since I couldn't find any reference to it perhaps good to add this example as a docstring in this else if path?

Reply:
Or alternatively, add a separate else if branch for ignored characters (e.g. "{"), just for clarity?

Reply:
It's in doc/descriptors.md but sure i'll expand a bit in a comment.